Improving recovery mechanism for long-running tasks #244407

doakalexi opened the pull request:
Resolves https://github.com/elastic/kibana/issues/244407

## Summary

This PR uses the frequent lease renewal strategy to allow safe task reclamation without waiting for long timeouts.

While long-running tasks are running, we update the task `retryAt` to now() + 5m every 1m.



### Checklist

- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios


### To verify

1. Update `...workflows_execution_engine/server/step/wait_step/wait_step.ts` to sleep for 15s
```
+ function sleep() {
+   return new Promise((resolve) => setTimeout(resolve, 15000));
+ }

export class WaitStepImpl implements NodeImplementation {
  constructor(
    private node: WaitGraphNode,
    private stepExecutionRuntime: StepExecutionRuntime,
    private workflowRuntime: WorkflowExecutionRuntimeManager,
    private workflowLogger: IWorkflowEventLogger
  ) {}

  async run(): Promise<void> {
  +   await sleep();
```
2. Start Kibana and ES
3. Enable workflows in [dev tools](http://localhost:5601/app/dev_tools#/console/shell)
```

POST kbn://internal/kibana/settings
{
  "changes": {
    "workflows:ui:enabled": true
  }
}
```
4. Open a new tab and go to the [workflows](http://localhost:5601/app/workflows) in stack management
5. Create a new workflow with a wait step
```
name: Test Demo
enabled: true
triggers:
 - type: manual
steps:
 - name: wait_step
   type: wait
   with:
    duration: 5s
```
6. Save your workflow and run it
7. While your workflow is running, run the following query to check the workflow task. Verify that the task retry value is being updated to `now() + 5m`.
```
GET /.kibana_task_manager/_search
{
  "query": {
    "bool": {
      "filter": [
        {
          "term": {
            "task.taskType": {
              "value": "workflow:run"
            }
          }
        }
      ]
    }
  },
  "_source": [
    "task.retryAt"
  ]
}
```
You should also see a log message at the start of the updates and when the task is finished running. I wonder if this is too much logging, bc I realized each workflow step is a long running task. Let me know what you think? Thanks!

doakalexi committed:
Updating task runner to update retryAt on interval for long running tasks

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/369203)
* Commit: 0c98fef4f027273495fc4cfff4eef8ff66e0c965

### Failed CI Steps
* [FTR Configs #7](https://buildkite.com/elastic/kibana-pull-request/builds/369203#019aeaf8-654e-4937-8a1b-f96267bc4e95)
* [FTR Configs #7](https://buildkite.com/elastic/kibana-pull-request/builds/369203#019aeb22-afeb-4c53-a81a-e9849cbb9b8f)
* [FTR Configs #52](https://buildkite.com/elastic/kibana-pull-request/builds/369203#019aeaf8-657a-4d1d-bd46-4d3faa86e56a)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/369203#019aeaf8-657a-4d1d-bd46-4d3faa86e56a) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/369203/jobs/019aeaf8-657a-4d1d-bd46-4d3faa86e56a/artifacts/019aeb20-f20b-408e-bebe-06dc56949e31) FTR Configs #<span></span>52 / Alerting improving alert severity should correctly set severity_improving and previous_action_group data in alert document
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/369203#019aeaf8-654e-4937-8a1b-f96267bc4e95) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/369203/jobs/019aeaf8-654e-4937-8a1b-f96267bc4e95/artifacts/019aeb22-3958-4bcd-8f64-79a10eb4ba9a) FTR Configs #<span></span>7 / Maps endpoints apis search ES|QL should return getValues response in expected shape
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/369203#019aeb22-afeb-4c53-a81a-e9849cbb9b8f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/369203/jobs/019aeb22-afeb-4c53-a81a-e9849cbb9b8f/artifacts/019aeb29-f6f3-4e90-8ee7-184b8ab984b9) FTR Configs #<span></span>7 / Maps endpoints apis search ES|QL should return getValues response in expected shape

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/369203","number":369203,"commit":"0c98fef4f027273495fc4cfff4eef8ff66e0c965"}],"number":369203}
buildkite-pr-comment-->

doakalexi committed:
Adding test

doakalexi committed:
Merge branch 'main' into task-manager/244407

doakalexi committed:
Merge branch 'main' into task-manager/244407

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/369261)
* Commit: c19bc8cf4ceb8c51e19d6fae728a06d09e7d2a64

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/369261#019aeb82-e063-4e60-ad5d-9fc181784476)

### History
* :broken_heart: [Build #369245](https://buildkite.com/elastic/kibana-pull-request/builds/369245) failed 56cbab742b66117ed3ecbc28a9235b81d16c4639
* :broken_heart: [Build #369203](https://buildkite.com/elastic/kibana-pull-request/builds/369203) failed 0c98fef4f027273495fc4cfff4eef8ff66e0c965

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/369261","number":369261,"commit":"c19bc8cf4ceb8c51e19d6fae728a06d09e7d2a64"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/369245","number":369245,"commit":"56cbab742b66117ed3ecbc28a9235b81d16c4639"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/369203","number":369203,"commit":"0c98fef4f027273495fc4cfff4eef8ff66e0c965"}],"number":369261}
buildkite-pr-comment-->

doakalexi committed:
Merge branch 'main' into task-manager/244407

doakalexi committed:
Merge branch 'main' into task-manager/244407

doakalexi committed:
Merge branch 'main' into task-manager/244407

doakalexi committed:
Adding functional test

doakalexi committed:
Merge branch 'task-manager/244407' of github.com:doakalexi/kibana into task-manager/244407

doakalexi committed:
Fixing test

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/370323)
* Commit: 438a8c967f06a8f17bdf79c33afa1daf67afac91

### Failed CI Steps
* [FTR Configs #21](https://buildkite.com/elastic/kibana-pull-request/builds/370323#019b0107-4deb-4e4d-b602-c72027ce6413)
* [FTR Configs #21](https://buildkite.com/elastic/kibana-pull-request/builds/370323#019b012a-548a-4dfd-9187-61c03183294a)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/370323#019b0107-4deb-4e4d-b602-c72027ce6413) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/370323/jobs/019b0107-4deb-4e4d-b602-c72027ce6413/artifacts/019b012a-244d-4554-9f43-2259f2900b87) FTR Configs #<span></span>21 / task_manager scheduling and running tasks should update the retryAt of a long running task to be now + 1m
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/370323#019b012a-548a-4dfd-9187-61c03183294a) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/370323/jobs/019b012a-548a-4dfd-9187-61c03183294a/artifacts/019b0135-5855-4fab-a506-d12f1ee9fbc5) FTR Configs #<span></span>21 / task_manager scheduling and running tasks should update the retryAt of a long running task to be now + 1m

### History
* :broken_heart: [Build #370302](https://buildkite.com/elastic/kibana-pull-request/builds/370302) failed b285314599c4d47f5d2732684fda157e48d83e48
* :broken_heart: [Build #369261](https://buildkite.com/elastic/kibana-pull-request/builds/369261) failed c19bc8cf4ceb8c51e19d6fae728a06d09e7d2a64
* :broken_heart: [Build #369245](https://buildkite.com/elastic/kibana-pull-request/builds/369245) failed 56cbab742b66117ed3ecbc28a9235b81d16c4639
* :broken_heart: [Build #369203](https://buildkite.com/elastic/kibana-pull-request/builds/369203) failed 0c98fef4f027273495fc4cfff4eef8ff66e0c965

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/370323","number":370323,"commit":"438a8c967f06a8f17bdf79c33afa1daf67afac91"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/370302","number":370302,"commit":"b285314599c4d47f5d2732684fda157e48d83e48"},{"buildStatus":{"state":"running","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/369261","number":369261,"commit":"c19bc8cf4ceb8c51e19d6fae728a06d09e7d2a64"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/369245","number":369245,"commit":"56cbab742b66117ed3ecbc28a9235b81d16c4639"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/369203","number":369203,"commit":"0c98fef4f027273495fc4cfff4eef8ff66e0c965"}],"number":370323}
buildkite-pr-comment-->

doakalexi committed:
Merge branch 'main' into task-manager/244407

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/370731)
* Commit: e83eca1cea04bdbd6e77792a76aca036f3cf6232

### Failed CI Steps
* [FTR Configs #13](https://buildkite.com/elastic/kibana-pull-request/builds/370731#019b0448-1d54-4644-a7f1-d9db87bff931)
* [FTR Configs #13](https://buildkite.com/elastic/kibana-pull-request/builds/370731#019b046f-8f30-405a-88ce-7b7badc1cb2d)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/370731#019b0448-1d54-4644-a7f1-d9db87bff931) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/370731/jobs/019b0448-1d54-4644-a7f1-d9db87bff931/artifacts/019b046f-25e0-45d7-ba8a-d64910cb04bb) FTR Configs #<span></span>13 / task_manager check_registered_task_types should check changes on all registered task types
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/370731#019b046f-8f30-405a-88ce-7b7badc1cb2d) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/370731/jobs/019b046f-8f30-405a-88ce-7b7badc1cb2d/artifacts/019b047c-d68d-4af8-8b5b-15aa7e3d9f55) FTR Configs #<span></span>13 / task_manager check_registered_task_types should check changes on all registered task types

### History
* :broken_heart: [Build #370323](https://buildkite.com/elastic/kibana-pull-request/builds/370323) failed 438a8c967f06a8f17bdf79c33afa1daf67afac91
* :broken_heart: [Build #370302](https://buildkite.com/elastic/kibana-pull-request/builds/370302) failed b285314599c4d47f5d2732684fda157e48d83e48
* :broken_heart: [Build #369261](https://buildkite.com/elastic/kibana-pull-request/builds/369261) failed c19bc8cf4ceb8c51e19d6fae728a06d09e7d2a64
* :broken_heart: [Build #369245](https://buildkite.com/elastic/kibana-pull-request/builds/369245) failed 56cbab742b66117ed3ecbc28a9235b81d16c4639
* :broken_heart: [Build #369203](https://buildkite.com/elastic/kibana-pull-request/builds/369203) failed 0c98fef4f027273495fc4cfff4eef8ff66e0c965

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/370731","number":370731,"commit":"e83eca1cea04bdbd6e77792a76aca036f3cf6232"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/370323","number":370323,"commit":"438a8c967f06a8f17bdf79c33afa1daf67afac91"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/370302","number":370302,"commit":"b285314599c4d47f5d2732684fda157e48d83e48"},{"buildStatus":{"state":"running","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/369261","number":369261,"commit":"c19bc8cf4ceb8c51e19d6fae728a06d09e7d2a64"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/369245","number":369245,"commit":"56cbab742b66117ed3ecbc28a9235b81d16c4639"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/369203","number":369203,"commit":"0c98fef4f027273495fc4cfff4eef8ff66e0c965"}],"number":370731}
buildkite-pr-comment-->

doakalexi committed:
Fixing test failure

doakalexi committed:
Merge branch 'task-manager/244407' of github.com:doakalexi/kibana into task-manager/244407

doakalexi committed:
Merge branch 'main' into task-manager/244407

doakalexi committed:
Updating from PR feedback

doakalexi committed:
Merge branch 'task-manager/244407' of github.com:doakalexi/kibana into task-manager/244407

doakalexi commented:
Made some changes, https://github.com/elastic/kibana/pull/245296/commits/f68763e9a74ffcea2d8d1411fb1b14ec87108129, from feedback:

- Moved stop timer function to `.finally` to make sure the timer is stopped on successful task run or error
- Added  stop timer function to `task.cancel` logic to make sure the timer is stopped when the task is canceled
- Updated the updateRetryAt logic to stop the timer when there is a `409` conflict error from `await this.bufferedTaskStore.partialUpdate`

doakalexi commented on x-pack/platform/plugins/shared/task_manager/server/task_running/task_runner.ts:
```diff
@@ -420,6 +429,11 @@ export class TaskManagerRunner implements TaskRunner {
       );
 
+      // Stop updating retryAt for long running tasks once the task has finished
+      if (stopUpdatingLongRunningTasks) {
```
Talked to @ymao1 and determined this change is not needed, as this PR doesn't affect any of the timeout/expiration logic. I did update the PR in this commit, https://github.com/elastic/kibana/commit/f68763e9a74ffcea2d8d1411fb1b14ec87108129, to make sure that the timer is stopped when a task times out.

doakalexi committed:
Updating for comments

doakalexi committed:
Merge branch 'main' into task-manager/244407

doakalexi committed:
Merge branch 'main' into task-manager/244407

ymao1 commented on x-pack/platform/plugins/shared/task_manager/server/lib/get_retry_at.test.ts:
```diff
@@ -87,3 +88,35 @@ describe('getRetryDate', () => {
+      createTaskRunner: jest.fn(),
+      timeout: '1d',
+    };
+    expect(getRetryAt(task, definition)).toEqual(new Date('2021-01-01T12:05:30.000Z'));
```
why would this be `2021-01-01T12:05:30`and not `2021-01-01T12:05:00`?

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/373247)
* Commit: 55cea1bc8772c61f53da442dfdb568e44b0b3cb6

### Failed CI Steps
* [FTR Configs #33](https://buildkite.com/elastic/kibana-pull-request/builds/373247#019b22bc-f434-4884-b985-8448100d7f89)
* [FTR Configs #97](https://buildkite.com/elastic/kibana-pull-request/builds/373247#019b22bc-f47f-44ac-b329-64e257aaab90)
* [FTR Configs #97](https://buildkite.com/elastic/kibana-pull-request/builds/373247#019b22e2-6eba-4c43-8b62-096bc774423f)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/373247#019b22bc-f434-4884-b985-8448100d7f89) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/373247/jobs/019b22bc-f434-4884-b985-8448100d7f89/artifacts/019b22e4-465c-4c62-91dc-6c60aa7b1cfa) FTR Configs #<span></span>33 / alerting api integration security and spaces enabled - Group 4 Alerts alerts getGlobalExecutionKpi should return KPI only from the current space
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/373247#019b22bc-f47f-44ac-b329-64e257aaab90) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/373247/jobs/019b22bc-f47f-44ac-b329-64e257aaab90/artifacts/019b22e2-4f8c-4ac6-8d29-15af11836b89) FTR Configs #<span></span>97 / Package policies Package Policy - input package behavior should not have created any ES assets on install
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/373247#019b22e2-6eba-4c43-8b62-096bc774423f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/373247/jobs/019b22e2-6eba-4c43-8b62-096bc774423f/artifacts/019b22ec-185a-4599-b3b5-2137c9a2a778) FTR Configs #<span></span>97 / Package policies Package Policy - input package behavior should not have created any ES assets on install

### History
* :green_heart: [Build #372604](https://buildkite.com/elastic/kibana-pull-request/builds/372604) succeeded acb5364cb74dfda5a3df2ea3cca4937e68b69e4b
* :broken_heart: [Build #372122](https://buildkite.com/elastic/kibana-pull-request/builds/372122) failed a77c19da5a0520d49380e593cc542b5565f9aa8b
* :green_heart: [Build #371374](https://buildkite.com/elastic/kibana-pull-request/builds/371374) succeeded 30a82561b6c68ea5c03ef99a138991b0f87c9b1c
* :broken_heart: [Build #370731](https://buildkite.com/elastic/kibana-pull-request/builds/370731) failed e83eca1cea04bdbd6e77792a76aca036f3cf6232
* :broken_heart: [Build #370323](https://buildkite.com/elastic/kibana-pull-request/builds/370323) failed 438a8c967f06a8f17bdf79c33afa1daf67afac91
* :broken_heart: [Build #370302](https://buildkite.com/elastic/kibana-pull-request/builds/370302) failed b285314599c4d47f5d2732684fda157e48d83e48

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373247","number":373247,"commit":"55cea1bc8772c61f53da442dfdb568e44b0b3cb6"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372604","number":372604,"commit":"acb5364cb74dfda5a3df2ea3cca4937e68b69e4b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372122","number":372122,"commit":"a77c19da5a0520d49380e593cc542b5565f9aa8b"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/371374","number":371374,"commit":"30a82561b6c68ea5c03ef99a138991b0f87c9b1c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/370731","number":370731,"commit":"e83eca1cea04bdbd6e77792a76aca036f3cf6232"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/370323","number":370323,"commit":"438a8c967f06a8f17bdf79c33afa1daf67afac91"}],"number":373247}
buildkite-pr-comment-->

doakalexi committed:
Merge branch 'main' into task-manager/244407

doakalexi committed:
Merge branch 'main' into task-manager/244407

ymao1 commented on x-pack/platform/test/plugin_api_integration/test_suites/task_manager/task_management.ts:
```diff
@@ -1735,5 +1735,24 @@ export default function ({ getService }: FtrProviderContext) {
+        expect(isNaN(retryAt)).to.be(false);
+
+        expect(retryAt).to.be.greaterThan(now + 5 * 60 * 1000);
+        expect(retryAt).to.be.lessThan(now + 6.5 * 60 * 1000);
```
i guess it wouldn't be good to wait a minute for this test to update the retryAt again 🙈

doakalexi replied:
We can if you'd like me to! I am not sure the best way to test multiple updates

ymao1 replied:
It would be nice to be able to test that the `retryAt` is updated but not sure it's worth such a long running test.

doakalexi replied:
I can make a follow up issue to try and figure out a good test for this

ymao1 reviewed (commented):
I think this looks good! Just a few comments.

doakalexi committed:
Updating to use default timeout

doakalexi committed:
Merge branch 'task-manager/244407' of github.com:doakalexi/kibana into task-manager/244407

ymao1 reviewed (approved):
LGTM. Just some comments about logs.

doakalexi commented on x-pack/platform/plugins/shared/task_manager/server/task_running/task_runner.ts:
```diff
@@ -933,6 +947,49 @@ export class TaskManagerRunner implements TaskRunner {
+      if (!stopped) {
+        try {
+          // Set retryAt to now + 5m
+          const updatedRetryAt = new Date(Date.now() + 5 * 60 * 1000);
```
Talked offline and decided not to make this change, as there is not task configured with a timeout < 5m

doakalexi committed:
Updating logging

doakalexi commented on x-pack/platform/plugins/shared/task_manager/server/task_running/task_runner.ts:
```diff
@@ -933,6 +947,49 @@ export class TaskManagerRunner implements TaskRunner {
+      if (!stopped) {
+        try {
+          // Set retryAt to now + 5m
+          const updatedRetryAt = new Date(Date.now() + 5 * 60 * 1000);
```
Updated in this commit, [b70f4ed](https://github.com/elastic/kibana/pull/245296/commits/b70f4ed333928843224c1dee3676035d7e0f98ad)

darnautov reviewed (approved):
LGTM

doakalexi committed:
Fixing test

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/373537)
* Commit: 1eec3df8eb478ff6c9f4a1d89888f1a277052329

### Failed CI Steps
* [FTR Configs #20](https://buildkite.com/elastic/kibana-pull-request/builds/373537#019b2434-08b5-4d7a-8472-09dea4cbe22f)
* [FTR Configs #22](https://buildkite.com/elastic/kibana-pull-request/builds/373537#019b2434-08b7-45a6-9f28-5ae7ac7b7f57)
* [FTR Configs #22](https://buildkite.com/elastic/kibana-pull-request/builds/373537#019b2455-86ab-47ae-8bbe-ba0bb5960677)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/373537#019b2434-08b5-4d7a-8472-09dea4cbe22f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/373537/jobs/019b2434-08b5-4d7a-8472-09dea4cbe22f/artifacts/019b245c-02da-49ee-82a8-7c06cf25c315) FTR Configs #<span></span>20 / Options list control Allow expensive queries setting is off Can search options list for available options - does not find partial match
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/373537#019b2455-86ab-47ae-8bbe-ba0bb5960677) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/373537/jobs/019b2455-86ab-47ae-8bbe-ba0bb5960677/artifacts/019b2466-4e02-4f7a-a72b-13525bf5156d) FTR Configs #<span></span>22 / serverless observability UI - Cases and Rules Serverless Observability Cases Cases list bulk actions status change the status of cases to in-progress correctly
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/373537#019b2434-08b7-45a6-9f28-5ae7ac7b7f57) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/373537/jobs/019b2434-08b7-45a6-9f28-5ae7ac7b7f57/artifacts/019b2455-524e-4816-86fa-21288dbbdee2) FTR Configs #<span></span>22 / serverless observability UI - Cases and Rules Serverless Observability Cases Cases list row actions Severity to high

### History
* :broken_heart: [Build #373449](https://buildkite.com/elastic/kibana-pull-request/builds/373449) failed b70f4ed333928843224c1dee3676035d7e0f98ad
* :broken_heart: [Build #373365](https://buildkite.com/elastic/kibana-pull-request/builds/373365) failed dcebea877475513ad5fd03f21d1e0b99dc3e278b
* :broken_heart: [Build #373247](https://buildkite.com/elastic/kibana-pull-request/builds/373247) failed 55cea1bc8772c61f53da442dfdb568e44b0b3cb6
* :green_heart: [Build #372604](https://buildkite.com/elastic/kibana-pull-request/builds/372604) succeeded acb5364cb74dfda5a3df2ea3cca4937e68b69e4b
* :broken_heart: [Build #372122](https://buildkite.com/elastic/kibana-pull-request/builds/372122) failed a77c19da5a0520d49380e593cc542b5565f9aa8b

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373537","number":373537,"commit":"1eec3df8eb478ff6c9f4a1d89888f1a277052329"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373449","number":373449,"commit":"b70f4ed333928843224c1dee3676035d7e0f98ad"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373365","number":373365,"commit":"dcebea877475513ad5fd03f21d1e0b99dc3e278b"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373247","number":373247,"commit":"55cea1bc8772c61f53da442dfdb568e44b0b3cb6"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372604","number":372604,"commit":"acb5364cb74dfda5a3df2ea3cca4937e68b69e4b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372122","number":372122,"commit":"a77c19da5a0520d49380e593cc542b5565f9aa8b"}],"number":373537}
buildkite-pr-comment-->

doakalexi committed:
Merge branch 'main' into task-manager/244407

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/373568)
* Commit: 2221c3e604545efdc7bf7fbd4e1f36835d3ada70

### Failed CI Steps
* [FTR Configs #26](https://buildkite.com/elastic/kibana-pull-request/builds/373568#019b248b-7bef-4da1-b517-5db08d65f3a7)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/373568#019b248b-7bef-4da1-b517-5db08d65f3a7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/373568/jobs/019b248b-7bef-4da1-b517-5db08d65f3a7/artifacts/019b24af-7bf1-4765-980a-c2ca074aca00) FTR Configs #<span></span>26 / security app useremail click changepassword link, change the password and re-login

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #373537](https://buildkite.com/elastic/kibana-pull-request/builds/373537) failed 1eec3df8eb478ff6c9f4a1d89888f1a277052329
* :broken_heart: [Build #373449](https://buildkite.com/elastic/kibana-pull-request/builds/373449) failed b70f4ed333928843224c1dee3676035d7e0f98ad
* :broken_heart: [Build #373365](https://buildkite.com/elastic/kibana-pull-request/builds/373365) failed dcebea877475513ad5fd03f21d1e0b99dc3e278b
* :broken_heart: [Build #373247](https://buildkite.com/elastic/kibana-pull-request/builds/373247) failed 55cea1bc8772c61f53da442dfdb568e44b0b3cb6
* :green_heart: [Build #372604](https://buildkite.com/elastic/kibana-pull-request/builds/372604) succeeded acb5364cb74dfda5a3df2ea3cca4937e68b69e4b
* :broken_heart: [Build #372122](https://buildkite.com/elastic/kibana-pull-request/builds/372122) failed a77c19da5a0520d49380e593cc542b5565f9aa8b

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373568","number":373568,"commit":"2221c3e604545efdc7bf7fbd4e1f36835d3ada70"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373537","number":373537,"commit":"1eec3df8eb478ff6c9f4a1d89888f1a277052329"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373449","number":373449,"commit":"b70f4ed333928843224c1dee3676035d7e0f98ad"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373365","number":373365,"commit":"dcebea877475513ad5fd03f21d1e0b99dc3e278b"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/373247","number":373247,"commit":"55cea1bc8772c61f53da442dfdb568e44b0b3cb6"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/372604","number":372604,"commit":"acb5364cb74dfda5a3df2ea3cca4937e68b69e4b"}],"number":373568}
buildkite-pr-comment-->
